app/models/git_repository.rb

```diff
@@ -0,0 +1,97 @@
+class GitRepository < ApplicationRecord
+  include AuthenticationMixin
+  validates :url, :format => URI::regexp(%w(http https)), :allow_nil => false
+  before_save :default_values
```

Comment:
You might want to use default_value_for over using a callback.

Also, can you put some whitespace line in here to break up the paragraphs...genererally, a blan line after the includes, one after the validates, one after the has_manysk